### PR TITLE
feat: new booker embed

### PR DESCRIPTION
**Pull request by JeroenReumkens**
## What does this PR do?

Make booker work with embed:
* Enables routes when cookie is set
* Gives an option in the embed configurator to select the layout you want to see
* No layout toggle in embed possible, we show the layout the user selects in the settings. 

Fixes CAL-1716

**Environment**: Staging(main branch)

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] View the embed playground in /packages/embed/embed-core by running `yarn dev`. 
- [ ] Test the embed settings on the event types (`<>` icon in top right when editing an event) to test new layout toggle

### CODE REVIEW

**Comment by hariombalhara** [comment]
Adding a single comment for all the feedback here.
- We should disable layout toggle buttons in embed atleast for `inline` mode. Having those buttons might break the inline embed experience. For popup embeds, it might be okay as Embed owns the entire screen in that case.
 image
- I think we should provide API to toggle layouts. Allowing that would let users configure embed layout based on his requirement. The API show also allow enabling/disabling the toggle buttons, so that website visitors can't do that themselves.
- For Popup embed, I think we should move the toggles close to the embed main content. It looks disconnected currently IMHO.
- Would it make sense to add Layout toggles on Routing Form as well for the case when Routing Form routes to an event.
https://www.loom.com/share/68a9f5fb70b049f39f56d1103061f851. Right now toggles aren't initially visible and come later when user routes to event page. Having the toggles on Routing Form allows him to give his preference earlier and open the routing form with a bigger layout(maybe?)

Bugs:
 - Too much blank space at the bottom and top here.
Screenshot 2023-05-24 at 12 30 42 PM
Screenshot 2023-05-24 at 12 31 12 PM
- New layouts don't have rounded borders
image
- Cal.com branding logo is missing on User Profile page. Might be the same with Team profile page. 
image
- Not sure if it's a bug in old booker as well but Paid event icon is there without any text for /paid event
image
- New Booker doesn't let it's borders configured. It was a feature requested not so long ago by a customer(I am not sure if he is using it right now)
https://www.loom.com/share/1afd7361abe548049f9e1b4aa8cdeac8

Some Observations:
- Align Left and Align Right options aren't working in new booker. I would prefer to not implement them in new booker as they are deprecated and I am not sure if someone is using them.

**Comment by hariombalhara** [comment]
Cal.com branding for new booker won't work for all backgrounds.
Here is how it looks on playground. We need to keep the background fixed for it to be visible in all cases.
Screenshot 2023-06-12 at 12 19 28 PM

**Comment by hariombalhara** [comment]
@JeroenReumkens Layout toggle is now working.
https://www.loom.com/share/135b967eb283484ea30ac3bd6571f58a

File: apps/web/components/Embed.tsx [hunk 1]
```diff
@@ -1070,6 +1091,27 @@ const EmbedTypeCodeAndPreviewDialogContent = ({
                         </div>
                       </Label>
                     ))}
+                    {isBookerLayoutsEnabled && (
```
Comment (JeroenReumkens) [inline apps/web/components/Embed.tsx:1094]: Adds the layout options to the embed settings popup.

File: apps/web/next.config.js [hunk 2]
```diff
@@ -224,12 +224,19 @@ const nextConfig = {
         destination: "/new-booker/:user/:type",
         has: [{ type: "cookie", key: "new-booker-enabled" }],
       },
+      {
```
Comment (JeroenReumkens) [inline apps/web/next.config.js:227]: Enables the urls for the embed.

File: apps/web/pages/new-booker/team/[slug]/[type].tsx [hunk 3]
```diff
@@ -14,8 +15,9 @@ import PageWrapper from "@components/PageWrapper";
 type PageProps = inferSSRProps<typeof getServerSideProps>;
 
 export default function Type({ slug, user, booking, away, isBrandingHidden }: PageProps) {
+  const isEmbed = typeof window !== "undefined" && window?.isEmbed?.();
   return (
-    <main className="flex h-full min-h-[100dvh] items-center justify-center">
+    <main className={classNames("flex h-full items-center justify-center", !isEmbed && "min-h-[100dvh]")}>
```
Comment (JeroenReumkens) [inline apps/web/pages/new-booker/team/[slug]/[type].tsx:20]: Small overflow / scroll tweaks.

File: packages/embeds/embed-core/index.html [hunk 4]
```diff
@@ -150,6 +150,12 @@ <h2>Popup Examples</h2>
         data-cal-link="forms/948ae412-d995-4865-875a-48302588de03">
         Book through Routing Form [Dark Theme]
       </button>
+      <button data-cal-namespace="popupPaidEvent" data-cal-config='{"layout":"week_view"}' data-cal-link="pro/paid">
```
Comment (JeroenReumkens) [inline packages/embeds/embed-core/index.html:153]: More examples in the embed palyground.

File: packages/features/bookings/Booker/Booker.tsx [hunk 5]
```diff
@@ -52,6 +57,14 @@ const BookerComponent = ({
   const extraDays = isTablet ? extraDaysConfig[layout].tablet : extraDaysConfig[layout].desktop;
   const bookerLayouts = event.data?.profile?.bookerLayouts || defaultBookerLayoutSettings;
   const animationScope = useBookerResizeAnimation(layout, bookerState);
+  const isEmbed = typeof window !== "undefined" && window?.isEmbed?.();
+  const isMonthView = layout === BookerLayouts.MONTH_VIEW;
+  // We only want the initial url value, that's why we memo it. The embed seems to change the url, which sometimes drops
+  // the layout query param.
+  const layoutFromQueryParam = useMemo(() => validateLayout(getQueryParam("layout") as BookerLayouts), []);
+  const defaultLayout = isEmbed
```
Comment (JeroenReumkens) [inline packages/features/bookings/Booker/Booker.tsx:65]: Embed won't use event settings, but rather will take the layout from the query param. That's set in the settings of the embed.

File: packages/features/bookings/Booker/components/Header.tsx [hunk 6]
```diff
@@ -35,7 +36,7 @@ export function Header({
     [setLayout, layout]
   );
 
-  if (isMobile || !enabledLayouts) return null;
+  if (isMobile || isEmbed || !enabledLayouts) return null;
```
Comment (JeroenReumkens) [inline packages/features/bookings/Booker/components/Header.tsx:39]: In the embed we don't show the layout toggles. You can't change layouts.

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**